Box.V2/Managers/BoxResourceManager.cs

```diff
@@ -225,6 +227,37 @@ protected async Task<BoxCollectionMarkerBasedV2<T>> AutoPaginateMarkerV2<T>(BoxR
             return allItemsCollection;
         }
 
+        /// <summary>
+        /// Used to fetch all results using pagination for metadata queries
+        /// </summary>
+        /// <typeparam name="T">The type of BoxCollectionMarkerBased item to expect.</typeparam>
+        /// <param name="request">The pre-configured BoxRequest object.</param>
+        /// <param name="limit">The limit specific to the endpoint.</param>
+        /// <returns></returns>
+        protected async Task<BoxCollectionMarkerBased<T>> AutoPaginateMarkerMetadataQuery<T>(BoxRequest request, int limit) where T : BoxMetadataQueryItem, new()
+        {
+            var allItemsCollection = new BoxCollectionMarkerBased<T>();
+            allItemsCollection.Entries = new List<T>();
+
+            bool keepGoing;
+            do
+            {
+                var response = await ToResponseAsync<BoxCollectionMarkerBased<T>>(request).ConfigureAwait(false);
+                var newItems = response.ResponseObject;
+                allItemsCollection.Entries.AddRange(newItems.Entries);
+                allItemsCollection.Order = newItems.Order;
+
+                dynamic body = JObject.Parse(request.Payload);
+                body.marker = newItems.NextMarker;
+                request.Payload = body.ToString();
```

Comment:
Except for these three lines, this function is almost the same as `AutoPaginateMarker<T>`.  Can you EASILY Refactor/Extract these three lines into a new method and consolidate the common code between the two functions?  If non-trival, then let's just keep it as-is.

Reply:
I initially tried to put it into a single method, but `AutoPaginateMarker<T>` expects `T` to be a `BoxEntity` and the metadata endpoint required creating a new model that was not a `BoxEntity`. So I created a new method for this case.